### ISSUE TITLE
Transformation of camera to Sky coordinates works only for MC

### ISSUE THREAD

**Issue by JouvinLea** [cta-observatory/cta-lstchain#420]
Hola,
Today, we realize that the utils method camera_to_sky in utils (https://github.com/cta-observatory/cta-lstchain/blob/master/lstchain/reco/utils.py) works only for MC since the obstime is hard coded.
I propose to do a small PR to fix the bug adding a time for the reference frame.
 Is it ok to replace the existing method ?

We could also add directly in the DL2 generation two columns ra and dec for the events!

**Comment by MaxNoe** [comment]
`camera_to_sky` transforms from `CameraFrame` to `HorizonFrame`, for which no obstime is needed.

It's actually badly named, as "sky" is not really clear, it should be `horizon` or `altaz`.

No obstime should be needed for processing simulations. Why was this hard coded in the first place? You don't need it for simulations and for observations it's wrong.

**Comment by rlopezcoto** [comment]
when you define `altaaz` you need to give an `obstime`, no?
https://github.com/cta-observatory/cta-lstchain/blob/6f98711c8bb7464ac3781abc0b9de054477eefad/lstchain/reco/utils.py#L47

this way you can directly transform `AltAz` to sky coordinates, or am I misunderstanding something?

**Comment by MaxNoe** [comment]
This works perfectly fine for mc:

```
pointing = SkyCoord(alt=alt, az=az, frame=AltAz())
camera_coord = SkyCoord(x=x, y=y, frame=CameraFrame(telescope_pointing=pointing, focal_length=focal_length)
altaz_coord = camera_coord.transform_to(AltAz())
```

Transformations from `CameraFrame` up to `AltAz`, including `TelescopeFrame` and `NominalFrame` do not neither require `location` nor `obstime`. 

For our current simulations, there is no well-defined way to go from `AltAz` to sky fixed positions, e.g. Equatorial Coordinates (ICRS) or Galactic Coordinates.
This might change if you produce gamma simulations based on a trajectory, but then you'd also have a well-defined location and obstime for simulations or if you simulate a star field for noise, in which case you could also do it by including a proper set of obstime / location in the mc or just provide pointing positions both in horizontal and in equatorial/galactic frame.

**Comment by rlopezcoto** [comment]
Thanks @MaxNoe. We can implement it as you are suggesting for MC, probably changing the misleading name of the function `camera_to_sky`, and add the transformation to sky coordinates for real data. Not sure if this goes against any data model requirement, but it would be great to have this information in the DL2 files as @JouvinLea was suggesting, any opinions? 
@JouvinLea could you take care of implementing this?

**Comment by MaxNoe** [comment]
Which information? Reconstructed position in sky coordinates?

**Comment by JouvinLea** [comment]
yes I will take care of it!!! I am +1 also to have it as optional argument, perfect.
I do a PR for that and then another one for DL2.

**Comment by moralejo** [comment]
If we want to have something like RA, dec for MC (which is "timeless"), we can use hour angle HA and declination.  They can be calculated from AltAz with just the observation location, which is indeed defined for any simulation (because of magnetic field & altitude, which enter at the Corsika level). 
Even for "tailored" MC with a given source path in the sky, the time would be irrelevant, the MC would be the same for any source at a given declination.

**Comment by rlopezcoto** [comment]
Solved in #421
